app/WebParsing/UtsgJsonParser.hs

```diff
@@ -42,17 +31,53 @@ insertAllMeetings org = do
     manager <- liftIO $ newManager tlsManagerSettings
     response <- liftIO $ httpLbs request' manager
     let respBody = responseBody response
+    return respBody
 
-    -- decode the response
+-- Get the page number, page size and total number of courses from response
+getPageInfo :: ByteString -> Maybe (Int, Int, Int)
+getPageInfo respBody = do
+    json <- decode respBody
+    flip parseMaybe json $ \obj -> do
+      payload <- obj .: "payload"
+      pageableCourse <- payload .: "pageableCourse"
+      page <- pageableCourse .: "page"
+      pageSize <- pageableCourse .: "pageSize"
+      totalCourses <- pageableCourse .: "total"
+      return (page, pageSize, totalCourses)
+          
+-- Helper function to insert courses for a page of a HTTP response
+insertCourses :: ByteString -> SqlPersistM ()
+insertCourses respBody = do
     let meetings :: Maybe DBList = decode respBody
     case meetings of
       Nothing -> return ()
-      Just dblist ->  mapM_ insertMeeting $ flattenDBList dblist
+      Just dblist -> mapM_ insertMeeting $ flattenDBList dblist
 
 -- | Helper function to flatten the list of DB Objects
 flattenDBList :: DBList -> [MeetTime]
 flattenDBList (DBList meetings) = concatMap (\(DB meetTimes) -> meetTimes) meetings
 
+-- | insert/update all the data into the Meeting and Times schema by creating and sending
+--   the http request to Artsci Timetable and then parsing the JSON response
+insertAllMeetings :: Int -> SqlPersistM ()
+insertAllMeetings page = do
+    respBody <- liftIO $ makeRequest page
+
+    let pageInfo :: Maybe (Int, Int, Int) = getPageInfo respBody
+    case pageInfo of 
+      Nothing -> return ()
+      Just (page', pageSize, totalCourses) -> do
```

Comment:
With this updated approach, `page'` is the same as `page`, and it's more explicit to just use `page` throughout. Here, you can use `_` to throw away the first value of the tuple